server/controllers/chefsController.js

```diff
@@ -3,9 +3,47 @@ const chefSchema = require("../models/chef");
 
 const Chef = connection.model("Chef", chefSchema);
 
+const findOneWithId = async (id) => {
+    const { _doc } = (await Chef.findById(id)) || {};
+    return _doc;
+};
+
+const findAllChef = async (query) => {
+    const chefs = await Chef.find(query);
+    return chefs;
+};
+
+const findOneWithUserId = async (userId) => {
+    const { _doc } = (await Chef.findOne({ userId })) || {};
+    return _doc;
+};
+
```

Comment:
What's the difference between this one and findOneWithId?

Reply:
This one finds the chef for a user's id and the `findOneWithId` finds the chef for chef's id